### ISSUE TITLE
Using scalars and func.count()

### ISSUE THREAD

**Issue by aurthurm** [absent1706/sqlalchemy-mixins#54]
I  want to peform a count of specific objects in the database. Currently i am doing it this way.   
`count = cls.where(patient_id__startswith='%P20%).count()`

But how do i do it with sqlalcmeny's `func.count()` function. I tries the line below but it does not work

`scalar = cls.query(func.count(Patient.uid)).filter(Patient.patient_id.like('P20')).scalar()`

where my Patient model extends the allfeaturemixin

```

@as_declarative()
class DBModel(AllFeaturesMixin):
    __name__: str
    __abstract__ = True
    
    uid = Column(Integer, primary_key=True, index=True, nullable=False, autoincrement=True)
    
    @declared_attr
    def __tablename__(cls) -> str:
        return cls.__name__.lower()


class Patient(DBModel):
    client_patient_id = Column(String, index=True, unique=True, nullable=False)
    patient_id = Column(String, index=True, unique=True, nullable=True)
    first_name = Column(String, nullable=False)
    middle_name = Column(String, nullable=True)
    last_name = Column(String, nullable=False)
    active = Column(Boolean(), default=True)
    
    @property
    def full_name(self):
        if self.middle_name:
            return f"{self.first_name} {self.middle_name} {self.last_name}"
        else:
            return f"{self.first_name} {self.last_name}"

    @classmethod
    def create(cls, obj_in: schemas.PatientCreate) -> schemas.Patient:
        data = cls._import(obj_in)
        return super().create(**data)

    def update(self, obj_in: schemas.PatientUpdate) -> schemas.Patient:
        data = self._import(obj_in)
        return super().update(**data) 
        
    @classmethod
    def create_patient_id(cls):
        prefix_key = "P"
        prefix_year = str(datetime.now().year)[2:]
        prefix = f"{prefix_key}{prefix_year}"
        count = cls.where(patient_id__startswith=f'%{prefix}%').count()
        # Find a way to use the line below: its faster
        scalar = cls.session.query(func.count(Patient.uid)).filter(Patient.patient_id.like(prefix)).scalar()
        if isinstance(count, type(None)):
            count = 0
        return f"{prefix}-{sequencer(count + 1, 5)}"
```

basically what i am trying to achieve here is to create patients with unique `patient_id` that follow a defined custom string sequence. The way i am implementing this is as follows

```
patient_id = models.Patient.create_patient_id()
incoming["patient_id"] = patient_id
patient = models.Patient.create(**incoming)
```

**Comment by Corcentric-Mbrown** [comment]
cls.where(some_predacate).select(func.count(1)).first() 

this works because the .select function is a sqlalchemy orm function not an extension, this builds the scalar into the select statement - its just read in reverse

**Comment by aurthurm** [comment]
Thanks @Corcentric-Mbrown i definitely missed that.
